ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -218,7 +218,8 @@ long getMaxBlobSize() {
           // delete is relevant
           deletedKeys.add(indexEntry.getKey());
         } else if (!isExpired(indexValue.getExpiresAtMs(), expirationReferenceTimeInMs)
-            && indexValue.getOriginalMessageOffset() != -1
+            && indexValue.getOriginalMessageOffset() != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET
+            && indexValue.getOriginalMessageOffset() != indexValue.getOffset().getOffset()
```

Comment:
Discussion in [this](https://github.com/linkedin/ambry/pull/608#discussion_r113773232) thread.